### ISSUE TITLE
Stackdriver Monitoring Exporter for Micrometer

### ISSUE THREAD

**Issue by saturnism** [spring-cloud/spring-cloud-gcp#1761]
Explore/add idiomatic integration w/ Stackdriver Monitoring and custom metrics. Typically in GKE deployments, users can use Micrometer and export metrics via Prometheus, and have Prometheus agents scrape & propagate metrics to Stackdriver Monitoring.

However, there are cases where it's better to propagate metrics directly to the service (e.g., when running in App Engine/serverless environment) to export metrics directly to Stackdriver Monitoring.

Micrometer can have custom exporters, it's possible to have a monitoring starter that automatically configure Micrometer and export to Stackdriver.

**Comment by bbzg** [comment]
Does anyone know the status of this? Is it still not provided out of the box? Thanks.

**Comment by meltsufin** [comment]
I haven't tired it, but there's [Stackdriver integration in the Micrometer project](https://github.com/micrometer-metrics/micrometer/tree/master/implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver).

**Comment by bbzg** [comment]
Yes I am using those, but I can't get them to "autowire" by just having them on the class path like other spring-cloud-gcp things. Would be nice if they could be detected and autowired. Unless someone already did that and I just missed some dependency...

**Comment by meltsufin** [comment]
We certainly don't have a Spring Boot Starter for it yet.
Contributions are welcomed!

**Comment by meltsufin** [comment]
Related: https://github.com/micrometer-metrics/micrometer-docs/pull/115

**Comment by eddumelendez** [comment]
there is currently support for `stackdriver` in `spring-boot` you can see the docs [here](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-metrics-export-stackdriver)

#2391 tries to solve the `project-id` and `credentials` auto-detection.
